src/randomInt.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Generate an inclusive random integer between `from` and `to`.
+ *
+ * @param from - The minimum value.
+ * @param to - The maximum value.
+ * @returns The random integer.
+ * @signature
+ *   R.randomInt(from, to)
+ * @example
+ *   R.randomInt(1, 10) // => 5
+ *   R.randomInt(1n, 10n) // => 7n
+ * @dataFirst
+ * @category Number
+ */
+export function randomInt<T extends bigint | number>(
+  from: T,
+  to: Widen<T>,
+): Widen<T> {
+  if (to <= from) {
+    throw new RangeError(`to(${to}) should be greater than from(${from})`);
+  }
+
+  if (typeof from === "bigint" && typeof to === "bigint") {
+    return BigInt(
+      Math.floor(Number(to - from + BigInt(1)) * Math.random() + Number(from)),
+    ) as never;
+  }
+
+  if (!Number.isFinite(from)) {
+    throw new TypeError(`from(${from}) is not a finite number`);
+  }
+
+  if (!Number.isFinite(to)) {
+    throw new TypeError(`to(${to}) is not a finite number`);
+  }
+
+  // Cast the values to numbers since TypeScript can't infer it properly.
+  const _from = from as number;
+  const _to = to as number;
+
+  if (Math.floor(_from) === Math.floor(_to)) {
+    throw new RangeError(
+      `there are no integers between from(${from}) and to(${to})`,
+    );
+  }
+
+  return Math.floor(Math.random() * (_to - _from + 1) + _from) as never;
```

Comment:
1. why are we casting as `never`?
2. Let's use the MDN implementation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_integer_between_two_values_inclusive

Reply:
See https://github.com/remeda/remeda/pull/793#discussion_r1704397718